### PR TITLE
Update `relation.units` docstring for subordinates

### PR DESCRIPTION
**Pull request by sed-i**
*Update `relation.units` docstring to mention the special case of subordinates units.*

## Checklist

 - [x] Have any types changed? If so, have the type annotations been updated?
 - [x] If this code exposes model data, does it do so thoughtfully, in a way that aids understanding?
 - [x] Do error messages, if any, present charm authors or operators with enough information to solve the problem?

## QA steps

N/A

## Documentation changes

No change.

## Bug reference

N/A.

## Changelog

- *Update `relation.units` docstring for subordinates*

### CODE REVIEW

File: ops/model.py [hunk 1]
```diff
@@ -1257,6 +1257,7 @@ class Relation:
         app: An :class:`Application` representing the remote application of this relation.
             For peer relations this will be the local application.
         units: A set of :class:`Unit` for units that have started and joined this relation.
+            For subordinate charms, this set will include one and only unit: the principal unit.
```
Comment (benhoyt) [inline ops/model.py:1260]: Thanks for this. I take it you've done your homework, but for the record, can you please point me to where in the code or documentation this is specified? Or is it a feature of Juju's `relation-list` hook tool?

```suggestion
            For subordinate charms, this set will include only one unit: the principal unit.
```

Comment (benhoyt) [reply ops/model.py]: Thanks. Can you please re-run `juju status` with `--relations` so we can see those details?

@jameinel Does this sound right, that relation-list should only list the principal unit when called from a subordinate charm?

Comment (sed-i) [reply ops/model.py]: > Can you please re-run `juju status` with `--relations` so we can see those details?

```
Model    Controller    Cloud/Region         Version  SLA          Timestamp
default  machineworld  localhost/localhost  2.9.38   unsupported  11:50:27-04:00

SAAS  Status      Store                URL
prom  terminated  charm-dev-batteries  admin/welcome.prom

App     Version  Status   Scale  Charm          Channel  Rev  Exposed  Message
kafka            active       1  kafka          edge     101  no       
uagent           blocked      4  grafana-agent            30  no       Missing relation: 'send-remote-write'
ubuntu  20.04    active       3  ubuntu         stable    21  no       
zk               active       1  zookeeper      edge      91  no       

Unit         Workload  Agent  Machine  Public address  Ports  Message
kafka/3*     active    idle   9        10.30.254.34           
  uagent/7   blocked   idle            10.30.254.34           Missing relation: 'send-remote-write'
ubuntu/2*    active    idle   12       10.30.254.221          
  uagent/5*  blocked   idle            10.30.254.221          Missing relation: 'send-remote-write'
ubuntu/3     active    idle   13       10.30.254.74           
  uagent/6   blocked   idle            10.30.254.74           Missing relation: 'send-remote-write'
ubuntu/4     active    idle   14       10.30.254.7            
  uagent/4   blocked   idle            10.30.254.7            Missing relation: 'send-remote-write'
zk/1*        active    idle   10       10.30.254.108          

Machine  State    Address        Inst id         Series  AZ  Message
9        started  10.30.254.34   juju-0728c2-9   jammy       Running
10       started  10.30.254.108  juju-0728c2-10  jammy       Running
12       started  10.30.254.221  juju-0728c2-12  focal       Running
13       started  10.30.254.74   juju-0728c2-13  focal       Running
14       started  10.30.254.7    juju-0728c2-14  focal       Running

Relation provider  Requirer          Interface   Type         Message
kafka:cluster      kafka:cluster     cluster     peer         
kafka:juju-info    uagent:juju-info  juju-info   subordinate  
kafka:restart      kafka:restart     rolling_op  peer         
ubuntu:juju-info   uagent:juju-info  juju-info   subordinate  
zk:cluster         zk:cluster        cluster     peer         
zk:restart         zk:restart        rolling_op  peer         
zk:zookeeper       kafka:zookeeper   zookeeper   regular
```

Comment (sed-i) [reply ops/model.py]: I suspect the correct wording would be
```diff
-For subordinate charms
+For subordinate relations
```

Also I discovered that when a subord. is related over CMRs, there can be more than one:
```
$ juju run --unit grafana-agent/0 -- relation-list -r 63           
prom/0
prom/1
```
```mermaid
graph LR

subgraph user-model in an lxd controller
subgraph zookeeper/0
zookeeper
grafana-agent
end
zookeeper --- grafana-agent

end

grafana-agent ---|CMR - send-remote-write| prom
grafana-agent ---|CMR - send-remote-write| prom2

subgraph cos-model in a K8s controller
subgraph prom
prom/0
prom/1
end

subgraph prom2
prom2/0
end

end

```

**Review by benhoyt — APPROVED** [review]
Thanks. This seems right to me based on the information you've given in the comments.

**Comment by benhoyt** [comment]
Given the above and the fact this is just a docstring change, merging without further review.

**Comment by sed-i** [comment]
For future reference, a more in-depth "investigation": https://github.com/canonical/grafana-agent-k8s-operator/issues/158#issuecomment-1487394310

**Pull request merged**